app/build.gradle.kts

```diff
@@ -0,0 +1,304 @@
+            assets.srcDir("$projectDir/schemas")
+        }
+    }
+
+    androidResources {
+        generateLocaleConfig = true
+    }
+
+    buildFeatures {
+        viewBinding = true
+        buildConfig = true
+    }
+
+    packaging {
+        resources {
+            // remove two files which belong to jsoup
+            // no idea how they ended up in the META-INF dir...
+            excludes += setOf(
+                "META-INF/README.md",
+                "META-INF/CHANGES",
+                "META-INF/COPYRIGHT" // "COPYRIGHT" belongs to RxJava...
+            )
+        }
+    }
+}
+
+val ktlint by configurations.creating
+
+checkstyle {
+    configDirectory = rootProject.file("checkstyle")
+    isIgnoreFailures = false
+    isShowViolations = true
+    toolVersion = libs.versions.checkstyle.get()
+}
+
+tasks.register<Checkstyle>("runCheckstyle") {
+    source("src")
+    include("**/*.java")
+    exclude("**/gen/**")
+    exclude("**/R.java")
+    exclude("**/BuildConfig.java")
+    exclude("main/java/us/shandian/giga/**")
+
+    classpath = configurations.getByName("checkstyle")
+
+    isShowViolations = true
+
+    reports {
+        xml.required = true
+        html.required = true
+    }
+}
+
+val outputDir = project.layout.buildDirectory.dir("reports/ktlint/")
+val inputFiles = fileTree("src") { include("**/*.kt") }
+
+tasks.register<JavaExec>("runKtlint") {
+    inputs.files(inputFiles)
+    outputs.dir(outputDir)
+    mainClass.set("com.pinterest.ktlint.Main")
+    classpath = configurations.getByName("ktlint")
+    args = listOf("src/**/*.kt")
+    jvmArgs = listOf("--add-opens", "java.base/java.lang=ALL-UNNAMED")
+}
+
+tasks.register<JavaExec>("formatKtlint") {
+    inputs.files(inputFiles)
+    outputs.dir(outputDir)
+    mainClass.set("com.pinterest.ktlint.Main")
+    classpath = configurations.getByName("ktlint")
+    args = listOf("-F", "src/**/*.kt")
+    jvmArgs = listOf("--add-opens", "java.base/java.lang=ALL-UNNAMED")
+}
+
+afterEvaluate {
+    tasks.named("preDebugBuild").configure {
+        if (!System.getProperties().containsKey("skipFormatKtlint")) {
+            dependsOn("formatKtlint")
+        }
+        dependsOn("runCheckstyle", "runKtlint", "checkDependenciesOrder")
+    }
+}
+
+sonar {
+    properties {
+        property("sonar.projectKey", "TeamNewPipe_NewPipe")
+        property("sonar.organization", "teamnewpipe")
+        property("sonar.host.url", "https://sonarcloud.io")
+    }
+}
+
+dependencies {
+    /** Desugaring **/
+    coreLibraryDesugaring(libs.android.desugar)
+
+    /** NewPipe libraries **/
+    implementation(libs.newpipe.nanojson)
+    // WORKAROUND: if you get errors with the NewPipeExtractor dependency, replace `v0.24.3` with
+    // the corresponding commit hash, since JitPack sometimes deletes artifacts.
+    // If there’s already a git hash, just add more of it to the end (or remove a letter)
+    // to cause jitpack to regenerate the artifact.
```

Comment:
This should reference libs.versions.toml

Reply:
I reverted this change in [`300f5ab` (#12684)](https://github.com/TeamNewPipe/NewPipe/pull/12684/commits/300f5abc70c081b9c09fa461b01531f5aee4b082)